src/keylessBackup/SignInWithEmail.tsx

```diff
@@ -90,15 +93,17 @@ function SignInWithEmailBottomSheet({
     </BottomSheet>
   )
 }
-
+type OAuthProvider = 'google-oauth2' | 'apple'
 type Props = NativeStackScreenProps<StackParamList, Screens.SignInWithEmail>
 
 function SignInWithEmail({ route }: Props) {
   const { t } = useTranslation()
   const dispatch = useDispatch()
+  const showApple = getFeatureGate(StatsigFeatureGates.SHOW_APPLE_IN_CAB)
   const { authorize, getCredentials, clearCredentials } = useAuth0()
   const { keylessBackupFlow, origin } = route.params
-  const [loading, setLoading] = useState(false)
+  const [loadingGoogle, setLoadingGoogle] = useState(false)
+  const [loadingApple, setLoadingApple] = useState(false)
```

Comment:
can we use a single state variable? with `null | OAuthProvider` as type?

Reply:
thats more elegant isn't it.